src/components/Ref/Ref.tsx

```diff
@@ -30,15 +41,39 @@ export default class Ref extends React.Component<RefProps> {
     innerRef: PropTypes.oneOfType([PropTypes.func, PropTypes.object]),
   }
 
+  static getDerivedStateFromProps(props: RefProps) {
+    const child = React.Children.only(props.children)
+
+    return {
+      child,
+      isForward: isForwardRef(child),
+    }
+  }
+
   componentDidMount() {
-    handleRef(this.props.innerRef, findDOMNode(this))
+    if (!this.state.isForward) {
+      handleRef(this.props.innerRef, findDOMNode(this))
+    }
   }
 
   componentWillUnmount() {
-    handleRef(this.props.innerRef, null)
+    if (!this.state.isForward) {
+      handleRef(this.props.innerRef, null)
+    }
+  }
+
+  private handleRefOverride = (node: HTMLElement) => {
```

Comment:
according to ref's docs: https://reactjs.org/docs/refs-and-the-dom.html

> The function receives the **React component instance** or HTML DOM element as its argument, which can be stored and accessed elsewhere.

The parameter type used in this function leads to false assumptions.

Reply:
See my comment below, we will enforce user to pass correct refs.